### ISSUE TITLE
Allow specifying more standard target triples

### ISSUE THREAD

**Issue by rib** [bbqsrc/cargo-ndk#79]
Hi,

I've been using cargo ndk for some time now and one thing that has been a little awkward at times is that `cargo ndk` uses ABI names (such as `armeabi-v7a`) as "target" names instead of standard triples like "armv7-linux-androideabi".

It's also a little misleading that the argument is referred to as a target and documented to be a "triple" (command line `--help` docs say: "triple for the target(s)")

One place where this is a bit awkward to work with is in defining github CI workflows where it can be desirable to setup a workflow matrix that would build a combination of operating systems, stable/nightly rust compilers and combination of targets. It's awkward when most other tools/components are using standard target triples but but `cargo ndk` uses ABI names and so it becomes necessary to map from one to the other.

Would you be open to updating `cargo ndk` so that the `-t`/`--target` arguments would accept target triples like `armv7-linux-androideabi` and `aarch64-linux-android`? I would think it could be ok for `cargo ndk` to continue also accepting ABI names to maintain backwards compatibility.

**Comment by bbqsrc** [comment]
Yup, feel free to submit a PR.

**Comment by bbqsrc** [comment]
This has actually been supported for a long time: https://github.com/bbqsrc/cargo-ndk/commit/9be44ab10a3f3f7ad1c15945b3490e61569a0e38.

**Comment by rib** [comment]
Yeah, I actually noticed this a few days when I came around to looking at this and was meaning to follow up here too. I suppose I must have found that it didn't work in the past and never tried to use triples again since https://github.com/bbqsrc/cargo-ndk/commit/9be44ab10a3f3f7ad1c15945b3490e61569a0e38.

I wonder if it'd be better to document the use of triples in the README so that this would be more apparent.

It's a bit of an aside but I also noticed that the reference to `"match rust triple architectures"` in the comment added in https://github.com/bbqsrc/cargo-ndk/commit/9be44ab10a3f3f7ad1c15945b3490e61569a0e38 also implies that these triples are somehow a Rust-specific notion which I don't think is really the case. I can recall these triples having a long history of use on Linux (probably first being used when cross compiling with gcc via autotools) and I think it would be fair to say they have become a very standard way of describing target architectures when cross compiling - predating Rust.

Either way though, happy that I should actually be able to switch over to using triples now. Thanks for the follow up to clarify that here.
